### DISCUSSION TITLE
flask db migrate fails on Initial migration

### DISCUSSION THREAD

**Discussion by cheizer** [miguelgrinberg/Flask-Migrate#467] [Q&A]
I found that flask migrate fails with a strange error ...

> schema_item.info = self.info.copy()
> AttributeError: 'str' object has no attribute 'copy'

When running flask db migrate. I narrowed the issue down to my model Column contains the "info" attribute. When I remove the "info" attribute, flask db migrate works fine.

**Comment by miguelgrinberg** [comment]
Isn't this a bug in your application? Instead of removing that attribute, why don't you fix the error? Python strings are immutable, they don't have to be copied, and for that reason there is no `copy()` method in them.

**Reply by cheizer** [reply]
I don't consider this a bug in my code as I did'nt do anything special, other than create a model and when defining the columns in the model I used the "info" attribute, which is documented.

This is my model

```
class CommonBase(db.Model):
	__abstract__ = True
	__tablelabel__ = ''

class TestTable(CommonBase):
	__tablename__ = 'test_table'

	rid             = Column(BigInteger, primary_key=True)
	name        = Column(String(255), nullable=False, info='Name')
```

**Reply by miguelgrinberg** [reply]
The documentation for `info` reads:

> **info** – Optional data dictionary which will be populated into the [SchemaItem.info](https://docs.sqlalchemy.org/en/14/core/metadata.html#sqlalchemy.schema.SchemaItem.info) attribute of this object.

Doesn't this mean that you are doing things wrong? You put a string in the `info` attribute.
